UnitySetup/UnitySetup.psm1

```diff
@@ -2225,7 +2226,11 @@ Would you like to continue? (Default: $($createPAT))"
 
         if ([string]::IsNullOrEmpty($azaccount) -or (-not $azaccount.Id.Contains("@microsoft.com"))) {
             Write-Verbose "Connecting to Azure, please login if prompted"
-            Connect-AzAccount | Out-Null
+            if ($AzureSubscription -ne [guid]::Empty) {
```

Comment:
Are you familiar with 'splatting' in PS? Could clean this code up as:

```powershell
$connectArgs = @{}
if ($AzureSubscription -ne [guid]::Empty) { 
    $connectArgs.Subscription = $AzureSubscription 
}
Connect-AzAccount @connectArgs  | Out-Null
```

'Splatting' makes it very nice to build up args to another call when they may or may not be specified. https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_splatting?view=powershell-7.4

Reply:
Yeah that's how I'm putting together the args to pass into `Update-UnityPackageManagerConfig` in the Mesh-side scripts, but I'm generally not very familiar with the right places to use it outside of situations where you need to pass a lot of args.

I'll give this a try, thanks!

Reply:
Yeah, any time I have 'optional' params that I'm sending into another cmdlet I'll usually do what you did here. Or if I've got a ton of them I'll do it too so it's easier to understand. Thanks for making the improvements!